docs/blog/posts/sled-integration.md

```diff
@@ -0,0 +1,248 @@
+related information was encoded in `/run/bpfd/programs/<ID>`. The extra infrastructure
+and failure modes introduced by this process was a constant headache, pushing the
+community to find a better solution.
+
+## Why Sled?
+
+[Sled](https://github.com/spacejam/sled) is an open source project described in
+github as "the champagne of beta embedded databases".  The "reasons" for choosing
+an embedded database from the [project website][sled] are pretty much spot on:
+
+```yaml
+Embedded databases are useful in several cases:
+
+- you want to store data on disk, without facing the complexity of files
+- you want to be simple, without operating an external database
+- you want to be fast, without paying network costs
+- using disk storage as a building block in your system
+```
+
+As discussed in the previous section, persistence across restarts, is one of
+bpfman's core design constraints, and with sled we *almost* get it for free!
+Additionally due to the pervasive nature of data management to bpfman's core
+workflow the data-store needed to be kept as simple and light weight as possible,
+ruling out heavier production-ready external database systems such as MySQL or
+Redis.
+
+Now this mostly focused on why embedded dbs in general, but why did we choose
+sled.....well because it's written in :crab: Rust :crab: of course! Apart from the
+obvious we took a small dive into the project before rewriting everything by
+[transitioning the OCI bytecode image library](https://github.com/bpfman/bpfman/pull/861)
+to use the db rather than the filesystem.  Overall the experience was extremely
+positive due to the following:
+
+- No more dealing directly with the filesystem, the sled instance is flushed to
+  the fs automatically every 500 ms by default and for good measure we manually
+  flush it before shutting down.
+- The API was extremly simple, traditional get and insert operations function
+  as expected.
+- Error handling with `sled:Error` is relatively simple and easy to map explicitly
+  to a `bpfmanError`
+- The db "tree" concept makes it easy to have separate keyspaces within the same
+  instance.
+
+## Transitioning to Sled
+
+Using the new embedded database started with the creation of a sled instance
+which could be easily shared across all of the modules in bpfman. To do this we
+utilized a [globally available `lazy_static` variable](hhttps://blog.logrocket.com/rust-lazy-static-pattern)
+called `ROOT_DB` in `main.rs`:
+
+```rust
+#[cfg(not(test))]
+lazy_static! {
+    pub static ref ROOT_DB: Db = Config::default()
+        .path(STDIR_DB)
+        .open()
+        .expect("Unable to open root database");
+}
+
+#[cfg(test)]
+lazy_static! {
+    pub static ref ROOT_DB: Db = Config::default()
+        .temporary(true)
+        .open()
+        .expect("Unable to open temporary root database");
+}
+```
+
+This block creates OR opens the filesystem backed database at `STDIR_DB`(/var/lib/bpfman/db)
```

Comment:
nit: formatting is a little odd here - you might omit the variable name and just spell out `/var/lib/bpfman/db`